ui/src/components/Investor/Investor.tsx

```diff
@@ -66,4 +95,41 @@ const Investor: React.FC<Props> = ({ onLogout }) => {
     </Switch>
 }
 
+const InvestorSetup: React.FC = () => {
+    const [ isPublic, setIsPublic ] = useState(true);
+    const [ loading, setLoading ] = useState(false);
+    const [ error, setError ] = useState('');
+
+    const investor = useParty();
+    const ledger = useLedger();
+
+    const handleSetupSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
```

Comment:
is this true for the issuer?

Reply:
The issuer (and all the roles, actually) will have to accept an `IssuerInvitation` (or `{Role}Invitation`) contract. The only difference is the investor's invite contract is the only one that has any data that has to be supplied...

I've been working on the Exchange view so I've been thinking about this, and maybe for the other roles, we can just open a `useStreamFetchByKey` to listen for the invitation, and auto-exercise the choice in the frontend. If more data is added at this stage later, we can extract it out into a form then